### DISCUSSION TITLE
Save multiple emails at the same time and save emails on user send email event, technical advice required

### DISCUSSION THREAD

**Discussion by C6900811** [OfficeDev/office-js#4023] [Q&A]
Dear Community,

We are working on a new M365 Outlook Add-in (Web Add-in) and we would like to do two things: 

**Save multiple emails at the same time from Outlook to our Platform via the Add-in** (user selects multiple emails via the UI). How can we fetch multiple emails at the same time through Graph API as .msg or .eml files on our platform? How can we get a list of selected emails from O365? Is there any way be which we pass the selected message ID and get the all the selected emails in single request?

**Save emails on "Send" event**, when user is writing a new email and sending it, we would like to save it to our Platform via the Add-in. Is there a way to do that? (it can be instant or asynchronous) 

We have made some technical research, but we couldn't find an adequate process to do it. Do you have any recommendations or advice? 

Thank you!!

**Comment by besiler** [comment]
Hi @C6900811 , we will help you find the best approach here. Would you mind copying this question into the Issues tab (https://github.com/OfficeDev/office-js/issues) since we comb that one regularly to find Outlook JS issues? Thanks!

**Reply by C6900811** [reply]
Thank you @besiler , we have added it here: https://github.com/OfficeDev/office-js/issues/4059
